osm-georeferencing/main.py

```diff
@@ -22,93 +26,209 @@ def start_kafka() -> None:
     producer = KafkaProducer(bootstrap_servers=[os.environ.get('KAFKA_PRODUCER_HOST')],
                              value_serializer=lambda m: json.dumps(m).encode('utf-8'))
     for msg in consumer:
-        try:
-            logging.info('Received message: ' + str(msg.value))
-            json_value = msg.value
-            specimen_data = json_value['data']
-            if specimen_data.get('dwc:locality') is None:
-                logging.warning('No locality information available, skipping message')
-            else:
-                result = run_georeference(specimen_data)
-                if result is not None:
-                    annotations = map_to_annotation(json_value, result)
-                    send_updated_opends(annotations, producer)
-        except:
-            logging.exception('Failed to process message')
-
-
-def map_to_annotation(json_value, result):
-    annotation_lat = {
-        'type': 'Annotation',
-        'motivation': 'https://hdl.handle.net/adding',
-        'creator': 'https://hdl.handle.net/enrichment-service-pid',
-        'created': timestamp_now(),
-        'target': {
-            'id': json_value['id'],
-            'type': 'https://hdl.handle.net/21...',
-            'indvProp': 'dwc:decimalLatitude'
-        },
-        'body': {
-            'source': result['display_name'],
-            'purpose': 'georeferencing',
-            'values': str(result['lat']),
-            'score': str(result['importance'])
+        logging.info('Received message: ' + str(msg.value))
+        json_value = msg.value
+        specimen_data = json_value['object']['digitalSpecimen']
+        result = run_georeference(specimen_data)
+        mas_job_record = map_to_mas_job_record(specimen_data, result, str(uuid.uuid4()))
+        send_updated_opends(mas_job_record, producer)
+
+
+def map_to_mas_job_record(specimen_data: Dict, results: List[Dict[str, str]], job_id: str) -> Dict:
+    """
+    Map the result of the API call to an annotation
```

Comment:
I think this needs to be 'map to a mas job record'?

Reply:
I meant to reference the description of the function rather than the actual function name